lib/afc/index.js

```diff
@@ -0,0 +1,237 @@
+    const res = await response;
+    if (res.opCode !== Operations.DATA) {
+      this.checkStatus(res);
+    }
+
+    return this.parseArray(res.content);
+  }
+
+  async openFile (path, mode) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const pathPayload = Buffer.from(path);
+    const fileModePayload = Buffer.alloc(8);
+    fileModePayload.writeUInt32LE(mode, 0);
+
+    const data = {
+      opCode: Operations.FILE_OPEN,
+      packetNumber,
+      headerPayload: Buffer.concat([fileModePayload, pathPayload])
+    };
+    this._encoder.write(data);
+    const res = await response;
+    if (res.opCode !== Operations.FILE_OPEN_RES) {
+      this.checkStatus(res);
+    }
+
+    return res.headerPayload.readUInt32LE(0);
+  }
+
+  async closeFileHandle (fileHande) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const fileModePayload = Buffer.alloc(8);
+    fileModePayload.writeUInt32LE(fileHande, 0);
+
+    const data = {
+      opCode: Operations.FILE_CLOSE,
+      packetNumber,
+      headerPayload: fileModePayload
+    };
+    this._encoder.write(data);
+    const res = await response;
+    this.checkStatus(res);
+  }
+
+  async fileWrite (fileHande, buffer) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const headerPayload = Buffer.alloc(8);
+    headerPayload.writeUInt32LE(fileHande, 0);
+
+    const data = {
+      opCode: Operations.FILE_WRITE,
+      packetNumber,
+      headerPayload,
+      content: buffer
+    };
+    this._encoder.write(data);
+    const res = await response;
+    this.checkStatus(res);
+  }
+
+  async fileRead (fileHande, length) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const headerPayload = Buffer.alloc(16);
+    headerPayload.writeUInt32LE(fileHande, 0);
+    headerPayload.writeUInt32LE(length, 8);
+
+    const data = {
+      opCode: Operations.FILE_READ,
+      packetNumber,
+      headerPayload
+    };
+    this._encoder.write(data);
+    const res = await response;
+    if (res.opCode !== Operations.DATA) {
+      this.checkStatus(res);
+    }
+    return res.content;
+  }
+
+  async getFileInfo (path) {
+    const {packetNumber, response} = this.createPacketPromise();
+
+    const data = {
+      opCode: Operations.GET_FILE_INFO,
+      packetNumber,
+      headerPayload: Buffer.from(path)
+    };
+    this._encoder.write(data);
+    const res = await response;
+    if (res.opCode !== Operations.DATA) {
+      this.checkStatus(res);
+    }
+    return this.parseObject(res.content);
+  }
+
+  checkStatus (res) {
+    if (res.opCode !== Operations.STATUS) {
+      throw new Error(`Unexpected response ${operationCode(res.opCode)}`);
+    }
+    if (_.isEmpty(res.headerPayload)) {
+      throw new Error('Header payload cant be empty for a status response');
+    }
+    if (res.headerPayload[0] !== Errors.SUCCESS) {
+      throw new Error(`Unexpected response ${errorCode(res.headerPayload[0])}`);
```

Comment:
do we show a number or the actual string description of the error code?

Reply:
The string description. `errorCode` function finds the representation of the error code